graphql2/graphqlapp/alert.go

```diff
@@ -42,33 +42,8 @@ func (a *AlertLogEntry) Message(ctx context.Context, obj *alertlog.Entry) (strin
 	return e.String(), nil
 }
 
-func (a *AlertLogEntry) escalationState(ctx context.Context, obj *alertlog.Entry) (*graphql2.AlertLogEntryState, error) {
-	e := *obj
-	meta, ok := e.Meta().(*alertlog.EscalationMetaData)
-	if !ok || meta == nil || !meta.NoOneOnCall {
-		return nil, nil
-	}
-
-	status := graphql2.AlertLogStatusWarn
-	return &graphql2.AlertLogEntryState{
-		Details: "No one was on-call",
-		Status:  &status,
-	}, nil
-}
-
-func (a *AlertLogEntry) notificationSentState(ctx context.Context, obj *alertlog.Entry) (*graphql2.AlertLogEntryState, error) {
-	e := *obj
-	meta, ok := e.Meta().(*alertlog.NotificationMetaData)
-	if !ok || meta == nil {
-		return nil, nil
-	}
-
-	s, err := (*App)(a).FindOneNotificationMessageStatus(ctx, meta.MessageID)
-	if err != nil {
-		return nil, errors.Wrap(err, "find alert log state")
-	}
-
-	var status graphql2.AlertLogStatus
+func GetNotificationState(s notification.MessageStatus) (*graphql2.NotificationState, error) {
```

Comment:
Since this is a helper method, let's not make it public. We can do a couple other things to make it clearer
- name it something declarative like `notificationStateFromStatus` instead of using `Get`
- remove the `error` from the return signature since it's not used/needed.

```suggestion
func notificationStateFromStatus(s notification.MessageStatus) *graphql2.NotificationState {
```

Reply:
when I removed the error from the return signature, I got an error from the return statement of "too many arguments to return"

Reply:
You'll need to update all the return statements to match